File: test-suite/src/test/java/io/micronaut/microstream/docs/DataRootProvider.java
package io.micronaut.microstream.docs;

import io.micronaut.microstream.conf.RootInstanceProvider;
import jakarta.inject.Named;
import jakarta.inject.Singleton;

@Named("one-microstream-instance") // <1>
@Singleton
public class DataRootProvider implements RootInstanceProvider<Data> {
    @Override
    public Data rootInstance() {
        return new Data();
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/DataRootProvider.kt
package io.micronaut.microstream.docs

import io.micronaut.microstream.conf.RootInstanceProvider
import jakarta.inject.Named
import jakarta.inject.Singleton

@Named("main")
@Singleton
class DataRootProvider : RootInstanceProvider<Data> {
    override fun rootInstance(): Data = Data()
}
